charts/consul/templates/client-daemonset.yaml

```diff
@@ -65,7 +65,7 @@ spec:
         {{- if .Values.global.secretsBackend.vault.agentAnnotations }}
         {{ tpl .Values.global.secretsBackend.vault.agentAnnotations . | nindent 8 | trim }}
         {{- end }}
-        {{- if .Values.global.enterpriseLicense.secretName }}
+        {{- if and .Values.global.enterpriseLicense.secretName (not .Values.global.acls.manageSystemACLs) }}
```

Comment:
changing it because license is not needed when ACLs are enabled because clients can pull it from servers.